Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -448,15 +453,27 @@ bool MediaPlayerPrivateGStreamerBase::handleSyncMessage(GstMessage* message)
             GST_TRACE("appending init data for %s of size %" G_GSIZE_FORMAT, eventKeySystemId, mapInfo.size);
             GST_MEMDUMP("init data", reinterpret_cast<const unsigned char *>(mapInfo.data), mapInfo.size);
             concatenatedInitDataChunks.append(mapInfo.data, mapInfo.size);
+#if USE(OPENCDM)
+            initDataChunks.append(mapInfo.data, mapInfo.size);
+#endif
             ++concatenatedInitDataChunksNumber;
             eventKeySystemIdString = eventKeySystemId;
             if (streamEncryptionInformation.second.contains(eventKeySystemId)) {
                 GST_TRACE("considering init data handled for %s", eventKeySystemId);
                 m_handledProtectionEvents.add(GST_EVENT_SEQNUM(event.get()));
+#if USE(OPENCDM)
+                keySystemProtectionEventMap.add(eventKeySystemId, GST_EVENT_SEQNUM(event.get()));
+#endif
             }
             gst_buffer_unmap(data, &mapInfo);
         }
+#if USE(OPENCDM)
+        Ref<SharedBuffer> initData = SharedBuffer::create(WTFMove(initDataChunks));
+        if (!(isInitDataCached(initData)))
+            m_initDataCount++;
 
+        m_initDataProtectionEventsMap.append(std::make_pair(WTFMove(initData), keySystemProtectionEventMap));
```

Comment:
If a protected content support more that one protection system, it will be better to keep systemid also along with event id. otherwise, we have to send session id to all event, even though not selected cdm system also. i.e, if a content support PR, WV and CK system, and WV is the selected cdm, if we send the event without checking event system id, we have to fire the events to PR and CK events also. So that we added system id also in the HashMap

Reply:
Ok, the problem I see is that you won't be able to store more than one event if you still use the `HashMap` with the sysid being the key. Maybe a `Vector<std::pair<uint32_t, String>>`.

Reply:
Ok understood, so we will update it

Reply:
Please see the reply above, since these are related comments